src/lf_apply.rs

```diff
@@ -52,70 +53,60 @@ unsafe fn backup_lpf<BD: BitDepth>(
             let top = (4 as c_int) << sb128;
```

Comment:
```suggestion
            let top = 4 << sb128;
```

This should let you avoid some of the extra casts below, I think.